LiteDB/Client/Mapper/BsonMapper.cs

```diff
@@ -30,7 +30,12 @@ public partial class BsonMapper
         /// Mapping cache between Class/BsonDocument
         /// </summary>
         private readonly Dictionary<Type, EntityMapper> _entities = new Dictionary<Type, EntityMapper>();
-
+        
+        /// <summary>
+        /// Unfinished mapping cache between Class/BsonDocument
+        /// </summary>
+        private readonly Dictionary<Type, EntityMapper> _buildEntities = new Dictionary<Type, EntityMapper>();
```

Comment:
how about using ConcurrentDictionary?

Reply:
We don't need concurrent access to this because it is only ever used in building the type mapping, which is always happening in the same thread.